cfme/tests/infrastructure/test_vm_reconfigure.py

```diff
@@ -81,21 +81,20 @@ def test_vm_reconfig_add_remove_hw_cold(
     # Disk modes cannot be specified when adding disk to VM in RHV provider
     lambda disk_mode, provider: disk_mode != 'persistent' and provider.one_of(RHEVMProvider))
 def test_vm_reconfig_add_remove_disk_cold(
-        provider, small_vm, ensure_vm_stopped, disk_type, disk_mode):
+        provider, small_vm, ensure_vm_stopped, disk_type, disk_mode, soft_assert):
 
     orig_config = small_vm.configuration.copy()
     new_config = orig_config.copy()
     new_config.add_disk(
-        size=5, size_unit='GB', type=disk_type, mode=disk_mode)
-
-    small_vm.reconfigure(new_config)
-    wait_for(
-        lambda: small_vm.configuration == new_config, timeout=360, delay=45,
-        fail_func=small_vm.refresh_relationships,
-        message="confirm that disk was added")
-
-    small_vm.reconfigure(orig_config)
-    wait_for(
-        lambda: small_vm.configuration == orig_config, timeout=360, delay=45,
-        fail_func=small_vm.refresh_relationships,
-        message="confirm that previously-added disk was removed")
+        size=500, size_unit='MB', type=disk_type, mode=disk_mode)
+
+    add_disk_request = small_vm.reconfigure(new_config)
+    wait_for(add_disk_request.is_succeeded, timeout=360, delay=45,
+             fail_func=small_vm.refresh_relationships,
+             message="confirm that disk was added")
+    soft_assert(lambda: small_vm.configuration == new_config, "Disk wasn't added to VM config")
```

Comment:
**REQUIRED**
I don't think `soft_assert` makes sense here. If a disk is not added, there will nothing to remove further.

Reply:
true.